stdpopsim/models.py

```diff
@@ -216,7 +281,21 @@ def get_samples(self, *args):
         """
         samples = []
         for pop_index, n in enumerate(args):
-            samples.extend([msprime.Sample(pop_index, time=0)] * n)
+            # If attempting to draw samples
+            if(n > 0):
+                # Only add to the samples vector if sampling is allowed
+                if(self.populations[pop_index].allow_samples):
+                    samples.extend(
+                        [msprime.Sample(
+                            pop_index,
+                            time=self.populations[pop_index].sampling_time
+                        )] * n
+                    )
+                else:
+                    raise ValueError("""Non-zero number of samples requested from
```

Comment:
This shouldn't be necessary if all sampling populations are adjacent now, right?

Reply:
Yeah, I can revert this to its original form or implement it with @apragsdale 's suggestion